addons/web/static/tests/views/form/form_view.test.js

```diff
@@ -1165,6 +1166,63 @@ test(`invisible fields are properly hidden`, async () => {
     expect(`.o_field_widget[name="child_ids"]`).toHaveCount(0);
 });
 
+test(`invisible fields are properly hidden (company evalContext)`, async () => {
+    Partner._records[0].int_field = 3;
+    serverState.companies = [
+        {
+            id: 1,
+            name: "Company 1",
+            sequence: 1,
+            parent_id: false,
+            child_ids: [],
+            country_code: "BE",
+        },
+        {
+            id: 2,
+            name: "Company 2",
+            sequence: 2,
+            parent_id: false,
+            child_ids: [],
+            country_code: "PE",
+        },
+        {
+            id: 3,
+            name: "Company 3",
+            sequence: 3,
+            parent_id: false,
+            child_ids: [],
+            country_code: "AR",
+        },
+    ];
+    cookie.set("cids", "3-1");
+
+    await mountView({
+        resModel: "partner",
+        type: "form",
+        // the arch contains an x2many field without inline view: as it is always invisible,
+        // the view should not be fetched. we don't specify any view in this test, so if it
+        // ever tries to fetch it, it will crash, indicating that this is wrong.
```

Comment:
I don't think this test needs to be so advanced (it's already tested). All you need to test is that you can use companies in an invisible (or required, or readonly) expression. Would be nice to test all keys though (the 3 variables and the function)

Reply:
edit: you test the eval context directly below, so probably no need to test each key here